widget/embedded/src/utils/time.ts

```diff
@@ -47,31 +55,42 @@ export function timeSince(millisecond: number): string {
   );
 
   for (const interval of sortedIntervals) {
-    const { turningPoint, label, pluralLabel } = interval;
+    const { turningPoint, label, pluralLabel, isExact } = interval;
     const intervalCount = Math.floor(seconds / turningPoint);
     if (intervalCount > 1) {
-      return `${intervalCount} ${pluralLabel}`;
-    }
-    if (intervalCount === 1) {
-      return `${intervalCount} ${label}`;
+      return {
+        time: isExact
+          ? `${new Date(millisecond).toLocaleDateString()} ${new Date(
+              millisecond
+            ).toLocaleTimeString()}`
+          : `${intervalCount} ${pluralLabel}`,
+        isExact,
+      };
+    } else if (intervalCount === 1) {
+      return {
+        time: isExact
+          ? `${new Date(millisecond).toLocaleDateString()} ${new Date(
+              millisecond
+            ).toLocaleTimeString()}`
+          : `${intervalCount} ${label}`,
+        isExact,
+      };
     }
   }
 
   if (seconds > 1) {
-    return `${seconds} ${i18n.t('seconds')}`;
+    return { time: `${seconds} ${i18n.t('seconds')}` };
   }
 
-  return `${seconds} ${i18n.t('second')}`;
+  return { time: `${seconds} ${i18n.t('second ago')}` };
 }
 
 export function getSwapDate(pendingSwap: PendingSwap) {
-  return pendingSwap.finishTime
-    ? i18n.t({
-        id: '{time} ago',
-        values: { time: timeSince(parseInt(pendingSwap.finishTime)) },
-      })
-    : i18n.t({
-        id: '{time} ago',
-        values: { time: timeSince(parseInt(pendingSwap.creationTime)) },
-      });
+  const { time, isExact } = pendingSwap.finishTime
+    ? timeSince(parseInt(pendingSwap.finishTime))
+    : timeSince(parseInt(pendingSwap.creationTime));
+  return i18n.t({
+    id: '{time}',
+    values: { time: isExact ? time : `${time} ${i18n.t('ago')}` },
+  });
```

Comment:
Consider updating the label for swap time on the swap page as well.
![image](https://github.com/rango-exchange/rango-client/assets/120931880/e58f7ef4-3575-49b5-bab0-c3817ee55782)

Reply:
fixed.